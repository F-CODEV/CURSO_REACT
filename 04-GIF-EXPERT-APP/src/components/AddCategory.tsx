import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }: AddCategoryProps) => {

    const [inputValue, setInputValue] = useState("");
    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        // console.log(e.target.value);
        setInputValue(e.target.value)

    }
    const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        // ? Agregamos en la primera posicion
        // setCategories((c) => [inputValue, ...c]);
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form onSubmit={(e) => onSubmit(e)} aria-label="form"
        >
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

export interface AddCategoryProps {
    // setCategories: React.Dispatch<React.SetStateAction<string[]>>
    onNewCategory: (newCategory: string) => void
}

