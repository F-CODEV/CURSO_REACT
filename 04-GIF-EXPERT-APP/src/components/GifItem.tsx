import { GifI } from "../interfaces/search_response.interface"

export const GifItem = ({ title, url }: GifI) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
