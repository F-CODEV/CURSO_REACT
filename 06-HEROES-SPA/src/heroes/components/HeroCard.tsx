import { Link } from "react-router";
import { HeroI } from "../data/heroes";
import './HeroCard.css';

export const HeroCard = ({
    id, superhero, alter_ego, publisher, characters, first_appearance
}: HeroI) => {

    return (
        <Link to={`/hero/${id}`} className="my-card animate__animated animate__fadeIn">
            <img
                src={`./assets/heroes/${id}.jpg`}
                className="img img-responsive"
                alt={superhero}
            />
            <div className="profile-name">{superhero}</div>
            <div className="profile-position">{alter_ego}</div>
            <div className="profile-overview">
                <div className="profile-overview">
                    <div className="row">
                        <div className="col-ms-4">
                            <h3>{publisher}</h3>
                            <p>Primera apariciÃ³n: <br />{first_appearance}</p>
                            {
                                (alter_ego !== characters)
                                && <p>{characters}</p>
                            }
                        </div>
                    </div>
                </div>
            </div>
        </Link>
    );
}
