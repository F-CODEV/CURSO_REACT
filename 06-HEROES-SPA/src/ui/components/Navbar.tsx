import { useContext } from 'react';
import { Link, NavLink, useNavigate } from 'react-router';
import { AuthContext } from '../../context/AuthContext';


export const Navbar = () => {

    const { user, logout } = useContext(AuthContext);
    const navigate = useNavigate();

    const onLogout = () => {
        logout();
        // ? replace para que no se pueda regresar con el boton de atras
        navigate('/login', { replace: true });
    }

    return (
        <nav
            className="navbar navbar-expand-lg bg-body-tertiary bg-dark"
            data-bs-theme="dark"
        >
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">Asociaciones</Link>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        {/* Si no necesitamos el isActive podemos usar Link */}
                        <NavLink
                            className={({ isActive }) => {
                                return `nav-link ${isActive ? 'active' : ''}`;
                            }}
                            to="/marvel" >
                            Marvel
                        </NavLink>
                        <NavLink
                            className={({ isActive }) => {
                                return `nav-link ${isActive ? 'active' : ''}`;
                            }}
                            to="/dc" >
                            DC
                        </NavLink>
                        <NavLink
                            className={({ isActive }) => {
                                return `nav-link ${isActive ? 'active' : ''}`;
                            }}
                            to="/search" >
                            Search
                        </NavLink>
                        <div className="d-flex">
                            <ul className="navbar-nav ml-auto">
                                <span className='nav-item nav-link text-primary'>{user?.name}</span>
                                <button
                                    aria-label='logout-btn'
                                    className="nav-item nav-link btn"
                                    onClick={onLogout}
                                >
                                    Logout
                                </button>
                            </ul>
                        </div>
                        {/* <NavLink
                            className={({ isActive }) => {
                                return `nav-link ${isActive ? 'active' : ''}`;
                            }}
                            to="/login" >
                            Logout
                        </NavLink> */}

                    </div>
                </div>
            </div>
        </nav>
    )
}