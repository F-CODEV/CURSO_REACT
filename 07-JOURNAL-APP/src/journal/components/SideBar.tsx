import { TurnedInNot } from "@mui/icons-material"
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"

export interface SideBarProps {
    drawerWidth: number
}
export const SideBar = ({ drawerWidth }: SideBarProps) => {
    return (
        <Box
            component='nav'
            sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}>
            <Drawer
                variant="permanent" //temporary si queremos ocultarlo de manera condicional
                open
                // open={true} para forma larga
                sx={{
                    display: { xs: 'block' },
                    '& .MuiDrawer-paper': { width: drawerWidth, boxSizing: 'border-box' }
                }}
            >
                <Toolbar>
                    <Typography variant="h6" noWrap component="div">
                        Francisco
                    </Typography>
                </Toolbar>
                <Divider />
                <List>
                    {['Enero', 'Febrero', 'Marzo', 'Abril'].map((e) => {
                        return <ListItem key={e} disablePadding>
                            <ListItemButton >
                                <ListItemIcon>
                                    <TurnedInNot />
                                </ListItemIcon>
                                <Grid container >
                                    <ListItemText

                                        primary={e}
                                        secondary={"Nostrud dolore aliqua mollit officia amet sit enim Lorem eiusmod cupidatat."}
                                    />
                                </Grid>
                            </ListItemButton>
                        </ListItem>
                    })}
                </List>

            </Drawer>

        </Box>
    )
}
