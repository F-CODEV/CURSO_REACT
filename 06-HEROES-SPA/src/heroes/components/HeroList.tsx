import { useMemo } from 'react';
import { getHeroesByPublihser } from '../helpers/getHeroesByPublihser';
import { HeroCard } from './HeroCard';

export interface HeroListProps {
    publisher: string;
}

export const HeroList = ({ publisher }: HeroListProps) => {
    // ? Cuando el publisher cambie se ejecutarÃ¡ la funcion
    const heroes = useMemo(() => getHeroesByPublihser(publisher), [publisher]);
    return (
        <div className='row rows-cols-1 row-cols-md-3 g-3'>
            {
                heroes.map(hero => (
                    <HeroCard
                        key={hero.id}
                        id={hero.id}
                        alter_ego={hero.alter_ego}
                        characters={hero.characters}
                        first_appearance={hero.first_appearance}
                        publisher={hero.publisher}
                        superhero={hero.superhero}
                    />
                ))
            }
        </div>
    )
}
