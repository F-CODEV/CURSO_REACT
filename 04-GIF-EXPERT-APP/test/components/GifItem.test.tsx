import { describe, expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import React from "react";
import { GifItem } from '../../src/components/GifItem';
import { GifI } from "../../src/interfaces/search_response.interface";

describe('Pruebas en GifItem.tsx', () => {

    const testGif: GifI = {
        id: "26gYOdBrT3MKcTj3O",
        title: "Good Boy Wolf GIF by League of Legends",
        url: "https://media1.giphy.com/media/26gYOdBrT3MKcTj3O/giphy.gif?cid=e87f0e7df7uklzp8zf6nft5bwz1sw3d021cmn86pa6weua55&ep=v1_gifs_search&rid=giphy.gif&ct=g"

    }

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem key={"a"} {...testGif} />);
        expect(container).toMatchSnapshot();
    });
    test('Debe mostrar la imagen con el URL y ALT indicado', () => {
        render(<GifItem key={"a"} {...testGif} />);
        // ? Para ver como esta el componente renderizado
        // screen.debug();
        const { src, alt } = screen.getByRole('img') as HTMLImageElement;
        expect(src).toBe(testGif.url);
        expect(alt).toBe(testGif.title);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem key={"a"} {...testGif} />);


        expect(screen.getByText(testGif.title)).toBeTruthy();

    });
});