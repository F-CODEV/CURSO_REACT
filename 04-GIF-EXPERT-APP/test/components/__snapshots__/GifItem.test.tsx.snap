// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem.tsx Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Good Boy Wolf GIF by League of Legends"
      src="https://media1.giphy.com/media/26gYOdBrT3MKcTj3O/giphy.gif?cid=e87f0e7df7uklzp8zf6nft5bwz1sw3d021cmn86pa6weua55&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Good Boy Wolf GIF by League of Legends
    </p>
  </div>
</div>
`;
